### PR TITLE
Update generated code

### PR DESCRIPTION
**Pull request by pjuarezd**
make generate-code
make regen-crd
make regen-crd-docs

### CODE REVIEW

File: docs/job_crd.adoc [hunk 1]
```diff
@@ -4,8 +4,8 @@
 [id="{p}-api-reference"]
 == API Reference
 
-:minio-image: https://hub.docker.com/r/minio/minio/tags[minio/minio:RELEASE.2024-03-15T01-07-19Z]
-:kes-image: https://hub.docker.com/r/minio/kes/tags[minio/kes:2024-03-13T17-52-13Z]
+:minio-image: https://hub.docker.com/r/minio/minio/tags[minio/minio:RELEASE.2024-05-01T01-11-10Z]
+:kes-image: https://hub.docker.com/r/minio/kes/tags[minio/kes:2024-04-12T13-50-00Z]
```
Comment (ramondeklein) [inline docs/job_crd.adoc:8]: I think we should also fix the `mc` image to a specific version (it's now set to latest). See also https://github.com/minio/operator/pull/2173#discussion_r1648645037.

File: docs/job_crd.adoc
Comment (ramondeklein) [inline docs/job_crd.adoc:1]: This file seems to be somehow duplicate https://github.com/minio/operator/pull/2173.

File: pkg/client/applyconfiguration/job.min.io/v1alpha1/commandspec.go [hunk 2]
```diff
@@ -1,5 +1,5 @@
 // This file is part of MinIO Operator
-// Copyright (c) 2023 MinIO, Inc.
+// Copyright (c) 2024 MinIO, Inc.
```
Comment (ramondeklein) [inline pkg/client/applyconfiguration/job.min.io/v1alpha1/commandspec.go:2]: Shouldn't this be 2023-2024?

Comment (pjuarezd) [reply pkg/client/applyconfiguration/job.min.io/v1alpha1/commandspec.go:2]: Should it?, it's an easy change but not sure, so far we keep a single year in header

Comment (ramondeklein) [reply pkg/client/applyconfiguration/job.min.io/v1alpha1/commandspec.go:2]: Okay, if we do it like this everywhere then it's fine. I'm just used to have that starting year in there as well. But that may be hard for generated files anyway...

File: docs/job_crd.adoc (hunk 1 above)
Comment (pjuarezd) [reply docs/job_crd.adoc:8]: mmmm, that is true, we are just pulling the latest MC and adding it on the crd doc, let me specify the mc version in the release.sh script, in a next PR

**Review by shtripat — APPROVED** [review]

File: docs/job_crd.adoc (hunk 1 above)
Comment (pjuarezd) [reply docs/job_crd.adoc:8]: set a specific mc image is covered in this PR @ramondeklein https://github.com/minio/operator/pull/2187

**Pull request merged**